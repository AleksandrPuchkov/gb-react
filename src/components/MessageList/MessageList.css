.message {
    list-style-type: none;
    margin: 20px 0px 20px 0px;
    padding: 10px;
    border-radius: 5px;
    width: 200px;
}

.usermessage {
    background-color: rgb(161, 234, 143);
    position: relative
}

.usermessage::before {
    content: "";
    position: absolute;
    display: block;
    transform: rotate(45deg);
    left: -5px;
    bottom: 8px;
    height: 10px;
    width: 10px;
    background-color: rgb(161, 234, 143);

}

.username,
.botname {
    font-size: 18px;
    font-weight: 700;
}

.username {
    color: rgb(114, 56, 214)
}

.usertext,
.bottext {
    font-weight: 100;
    font-size: 14px;
    white-space: nowrap;
}

.botmessage {
    position: relative;
    margin-left: 40%;
    background-color: rgb(240, 240, 240);
}

.botmessage::before {
    content: "";
    position: absolute;
    display: block;
    transform: rotate(45deg);
    left: 215px;
    bottom: 8px;
    height: 10px;
    width: 10px;
    background-color: rgb(240, 240, 240);

}