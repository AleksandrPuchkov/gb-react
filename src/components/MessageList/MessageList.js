"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageList = void 0;
const react_1 = __importStar(require("react"));
require("./MessageList.css");
exports.MessageList = (0, react_1.memo)(({ messages }) => (react_1.default.createElement("ul", null, messages.map((message, idx) => {
    if (message.username === 'Chatbot')
        return (react_1.default.createElement("li", { key: idx, className: "message botmessage" },
            react_1.default.createElement("h3", { className: "botname" }, message.username),
            react_1.default.createElement("p", { className: "bottext" }, message.message)));
    else
        return (react_1.default.createElement("li", { key: idx, className: "message usermessage" },
            react_1.default.createElement("h3", { className: "username" }, message.username),
            react_1.default.createElement("p", { className: "usertext", "data-testid": "testmessage" }, message.message)));
}))));
