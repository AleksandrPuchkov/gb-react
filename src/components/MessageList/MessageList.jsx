import React from 'react';
import './MessageList.css';

export const MessageList = ({ messages }) => (
  <ul>
    {messages.map((message, idx) => {
      if (message.username === 'Chatbot')
        return (
          <li key={idx} className="message botmessage">
            <h3 className="botname">{message.username}</h3>{' '}
            <p className="bottext">{message.message}</p>
          </li>
        );
      else
        return (
          <li key={idx} className="message usermessage">
            <h3 className="username">{message.username}</h3>{' '}
            <p className="usertext">{message.message}</p>
          </li>
        );
    })}
  </ul>
);
