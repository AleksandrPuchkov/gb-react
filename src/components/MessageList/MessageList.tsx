import React, { FC, memo } from 'react';
import { nanoid } from 'nanoid';
import './MessageList.css';

interface Message {
  id: string,
  username: string,
  message: string
}

interface MessageListProps {
  messages: Message[] // массив значений Message (объектов) || Array<Message>
}

export const MessageList: FC<MessageListProps> = memo(({ messages }) => (
  <ul>
    {messages.map((message, idx) => {
      if (message.username === 'Chatbot')
        return (
          <li key={idx} className="message botmessage">
            <h3 className="botname">{message.username}</h3>
            <p className="bottext">{message.message}</p>
          </li>
        );
      else
        return (
          <li key={idx} className="message usermessage">
            <h3 className="username">{message.username}</h3>
            <p className="usertext" data-testid="testmessage">
              {message.message}
            </p>
          </li>
        );
    })}
  </ul>
));
