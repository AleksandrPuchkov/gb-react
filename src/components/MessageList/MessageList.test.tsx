import React from 'react';
import { MessageList } from './MessageList';
import { render, screen } from '@testing-library/react';

const testmessages = [{ id: 'id', username: 'TestUser', text: 'TestMessage' }];

describe('MessageList', () => {
  it('Render component', () => {
    render(<MessageList messages={testmessages} />);
  });

  it('To match snapshot', () => {
    const { asFragment } = render(<MessageList messages={testmessages} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('To have username and message fields', () => {
    render(<MessageList messages={testmessages} />);
    expect(screen.getByRole('list')).toBeInTheDocument;
    expect(screen.getByRole('heading').innerHTML).toBe('TestUser');
    expect(screen.getByTestId('testmessage').innerHTML).toBe('TestMessage');
  });
});
