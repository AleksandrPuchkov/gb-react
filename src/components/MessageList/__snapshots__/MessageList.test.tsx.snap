// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList To match snapshot 1`] = `
<DocumentFragment>
  <ul>
    <li
      class="undefined undefined"
    >
      <h3>
        TestUser
      </h3>
      <p
        data-testid="testmessage"
      >
        TestMessage
      </p>
    </li>
  </ul>
</DocumentFragment>
`;
